iml-api/src/graphql/mod.rs

```diff
@@ -1193,3 +1161,71 @@ fn create_task_job<'a>(task_id: i32) -> SendJob<'a, HashMap<String, serde_json::
             .collect(),
     }
 }
+
+async fn get_targets(
+    pool: &PgPool,
+    limit: Option<i32>,
+    offset: Option<i32>,
+    dir: Option<SortDir>,
+    fs_name: Option<String>,
+    exclude_unmounted: bool,
+) -> juniper::FieldResult<Vec<Target>> {
+    let dir = dir.unwrap_or_default();
+
+    let xs: Vec<Target> = sqlx::query!(
+        r#"
+            SELECT state, name, active_host_id, host_ids, filesystems, uuid, mount_path, dev_path, fs_type::text from target t
+            ORDER BY
+                CASE WHEN $3 = 'asc' THEN t.name END ASC,
+                CASE WHEN $3 = 'desc' THEN t.name END DESC
+            OFFSET $1 LIMIT $2"#,
+        offset.unwrap_or(0) as i64,
+        limit.map(|x| x as i64),
+        dir.deref()
+    )
+    .fetch(pool)
+    .map_ok(|x| {
+        Target {
+            state: x.state,
+            name: x.name,
+            active_host_id: x.active_host_id,
+            host_ids: x.host_ids,
+            filesystems: x.filesystems,
+            uuid: x.uuid,
+            mount_path: x.mount_path,
+            dev_path: x.dev_path,
+            fs_type: x.fs_type.unwrap_or_else(|| "ldiskfs".to_string()).into()
```

Comment:
How can we retroactively assume what the fs_type is?

Reply:
I changed it to be optional.

Reply:
I changed it to be an option.